File: GROUP_13_2020510014_2020510025_2020510075.rb
def euclidean_distance(point1, point2)
  if point1.length != point2.length
    raise "Points must have the same dimension"
  end

  squared_diff_sum = point1.zip(point2).map { |a, b| (a - b) ** 2 }.sum
  Math.sqrt(squared_diff_sum)
end

point1 = [3.0, 1.0, 2.0]
point2 = [2.0, 4.0, 6.0]
puts "Euclidean Distance: #{euclidean_distance(point1, point2)}"

File: GROUP_13_2020510014_2020510025_2020510075.rs
fn euclidean_distance(point1: &[f64], point2: &[f64]) -> f64 {
    if point1.len() != point2.len() {
        panic!("Points must have the same dimension");
    }

    let squared_diff_sum: f64 = point1.iter()
        .zip(point2.iter())
        .map(|(a, b)| (a - b).powi(2))
        .sum();

    squared_diff_sum.sqrt()
}

fn main() {
    let point1 = vec![3.0, 1.0, 2.0];
    let point2 = vec![2.0, 4.0, 6.0];
    println!("Euclidean Distance: {}", euclidean_distance(&point1, &point2));
}
